packages/primitives/paragraph/src/Root.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { normalizeWebStyle, TStyle } from 'stili'
+import {
+  component,
+  mapDefaultProps,
+  mapWithProps,
+  startWithType,
+} from 'refun'
+import { colorToString, isColor } from 'colorido'
+import { TParagraph } from './types'
+
+export const Paragraph = component(
+  startWithType<TParagraph>(),
+  mapDefaultProps({
+    shouldPreserveWhitespace: false,
+    shouldPreventSelection: false,
+    shouldPreventWrap: false,
+    shouldHideOverflow: false,
+    isUnderlined: false,
+  }),
+  mapWithProps(({
+    align,
+    blockEnd,
+    blockStart,
+    color,
+    direction,
+    inlineEnd,
+    inlineStart,
+    letterSpacing,
+    lineHeight,
+    fontFamily,
+    fontWeight,
+    fontSize,
+    isUnderlined,
+    shouldPreserveWhitespace,
+    shouldPreventSelection,
+    shouldPreventWrap,
+    shouldHideOverflow,
+  }) => {
+    const style: TStyle = {
+      _webOnly: {
+        fontSmoothing: 'antialiased',
+        maxWidth: '100%',
+        textRendering: 'geometricPrecision',
+        textSizeAdjust: 'none',
+      },
+      fontFamily,
+      fontSize,
+      fontWeight,
+      minWidth: 0,
+      letterSpacing,
+      lineHeight,
+      paddingBottom: blockEnd,
+      paddingLeft: inlineStart,
+      paddingRight: inlineEnd,
+      paddingTop: blockStart,
+    }
+
+    if (isColor(color)) {
+      style.color = colorToString(color)
+    }
+
+    if (align) {
+      switch (align) {
+        case 'start': {
+          style.textAlign = direction === 'right-to-left' ? 'right' : 'left'
+
+          break
+        }
+
+        case 'center': {
+          style._webOnly!.textAlign = 'center'
+
+          break
+        }
+
+        case 'end': {
+          style.textAlign = direction === 'right-to-left' ? 'left' : 'right'
```

Comment:
if i wanna right in hebrew, i'll do `align === 'end'` and `direction === 'right-to-left` 
but than i'll get style.textAlign = left
🤔

Reply:
I thought that was what it was expected. Let me do a quick drawing:

<img width="773" alt="Screen Shot 2020-05-19 at 23 50 05" src="https://user-images.githubusercontent.com/381614/82382001-7e369900-9a2b-11ea-84ac-6c2f750c8292.png">

Does my logic make sense?

Reply:
wow! i'm not sure if its too smart for me...

lets see what the others think about it 🤔

Reply:
I also had some "big brain time" with this one to understand that when you change the `direction` then the `alignment` changes, because it's `START | center | END` and not `LEFT | center | RIGHT`, but I think it's great 👍